ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,32 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  if List.assoc "secureboot" platform = "auto" then
+    let t = ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) in
+    List.append [t] (List.remove_assoc "secureboot" platform)
+  else
+    platform
+
+let save_uefi_certificates_to_dir ~__context ~self =
+  let err () =
+    debug "Error writing uefi certificates to file";
+    failwith "Failed to save UEFI certificates: varstored directory does not exist or is not a directory" in
+
+  try
+    let uefi_platform_key = !Xapi_globs.varstore_dir ^ "PK.auth" in
+    if not (Sys.file_exists uefi_platform_key) then
+      if Sys.file_exists !Xapi_globs.varstore_dir && Sys.is_directory !Xapi_globs.varstore_dir then
+        let filename = "xapi_uefi_certificates.tar" in
+        Unixext.with_file filename [Unix.O_RDWR; Unix.O_CREAT] 0o755 (fun fd ->
+            let contents = (Xapi_stdext_base64.Base64.decode (Db.Pool.get_uefi_certificates ~__context ~self)) in
+            Unixext.write_string_to_file filename contents;
+            Tar_unix.Archive.extract (fun _ -> !Xapi_globs.varstore_dir) fd)
+      else
+        err ()
+  with Sys_error _ ->
```

Comment:
Is this try/with required here? I think you replaced all the Sys. function calls with Sys.file_exists which do not raise.